tests/test_default_browser.py

```diff
@@ -0,0 +1,110 @@
+# noqa: F401, F811
+# pylint: disable=redefined-outer-name,unused-argument,unused-import
+
+import webbrowser
+
+import pytest
+
+from browser_history import browsers, utils
+
+from .utils import become_mac, become_linux, become_windows  # noqa: F401
+
+
+platform = utils.get_platform()
+
+
+class MockBrowser:
+    """Mock version of return value of webbrowser.get()"""
+
+    def __init__(self, name=None):
+        self.name = name
+
+
+@pytest.fixture
+def change_linux_default(monkeypatch, request):
+    """Changes utils._default_browser_linux to return a specific named
+    browser. use @pytest.mark.browser_name(name) to set
+    browser name
+    """
+
+    marker = request.node.get_closest_marker("browser_name")
+
+    if marker is None:
+        browser_name = None
+    else:
+        browser_name = marker.args[0]
+
+    def mock_get():
+        return browser_name
+
+    monkeypatch.setattr(utils, "_default_browser_linux", mock_get)
+
+
+@pytest.fixture
+def change_win_default(monkeypatch, request):
+    """Changes utils._default_browser_win to return a specific named
+    browser. use @pytest.mark.browser_name(name) to set
+    browser name
+    """
+    if platform != utils.Platform.WINDOWS:
+        pytest.skip("Skipping windows registry based test")
+
+    marker = request.node.get_closest_marker("browser_name")
+
+    if marker is None:
+        browser_name = None
+    else:
+        browser_name = marker.args[0]
+
+    def mock_get():
+        return browser_name
+
+    monkeypatch.setattr(utils, "_default_browser_win", mock_get)
```

Comment:
The problem with testing it like this is that `_default_browser_linux` and `_default_browser_win` are left untested. Since the code in them is very platform-dependent, we should test them thoroughly (ex: if the process returns an empty output, etc.). So we could have separate tests for these.

I'll try to write these tests today.

Reply:
I think we need to test this deeply, without monkeypatching these complex steps like reading the registry and piping output from the linux command line. I don't know if it can be done.